python/tank/descriptor/io_descriptor/appstore.py

```diff
@@ -359,30 +356,13 @@ def download_local(self):
         #  'link_type': 'upload'}
         attachment_id = version[constants.TANK_CODE_PAYLOAD_FIELD]["id"]
 
-        # and now for the download.
-        # @todo: progress feedback here - when the SG api supports it!
-        # sometimes people report that this download fails (because of flaky connections etc)
-        # engines can often be 30-50MiB - as a quick fix, just retry the download once
-        # if it fails.
-        log.debug("Downloading attachment %s..." % self._version)
+        # download and unzip
         try:
-            bundle_content = sg.download_attachment(attachment_id)
-        except Exception, e:
-            # retry once
-            log.debug("Downloading failed, retrying. Error: %s" % e)
-            bundle_content = sg.download_attachment(attachment_id)
-
-        zip_tmp = os.path.join(tempfile.gettempdir(), "%s_tank.zip" % uuid.uuid4().hex)
-        fh = open(zip_tmp, "wb")
-        fh.write(bundle_content)
-        fh.close()
-
-        # unzip core zip file to app target location
-        log.debug("Unpacking %s bytes to %s..." % (os.path.getsize(zip_tmp), target))
-        unzip_file(zip_tmp, target)
-
-        # remove zip file
-        filesystem.safe_delete_file(zip_tmp)
+            shotgun.download_and_unpack_attachment(sg, attachment_id, target)
+        except ShotgunAttachmentDownloadError:
+            raise TankAppStoreError(
```

Comment:
This'll end up masking the real error, right?  Seems like we should at least include a description from the actual error.